### ISSUE TITLE
Problems with height? Node get wrong height?

### ISSUE THREAD

**Issue by MichaelHoelzl** [vislyhq/stretch#14]
```Rust
use stretch::geometry::Size;
use stretch::style::*;

fn main() {
    let node = Node {
        size: Size { width: Dimension::Points(800.0), height: Dimension::Points(600.0) },
        justify_content: JustifyContent::Center,

        children: vec![Node {
            size: Size { width: Dimension::Percent(0.5), height: Dimension::Percent(0.5) },
            ..Default::default()
        }],

        ..Default::default()
    };

    let layout = stretch::compute(&node, Size::undefined());

    println!("{:#?}", layout);
}
```

And the result... But doesn't the sub node have to get a height of 300?
I mean, 50 percent of 600 is 300. 😅 

```Code
Node {
    order: 0,
    size: Size {
        width: 800.0,
        height: 600.0
    },
    location: Point {
        x: 0.0,
        y: 0.0
    },
    children: [
        Node {
            order: 0,
            size: Size {
                width: 400.0,
                height: 400.0
            },
            location: Point {
                x: 200.0,
                y: 0.0
            },
            children: []
        }
    ]
}
```

**Comment by emilsjolander** [comment]
Good find! margin and padding percentages are based on width regardless of what edge the margin / padding is set on so this is how this mistake was made. Thanks for reporting!

**Comment by emilsjolander** [comment]
Should have a fix up in next day or so 👍

**Comment by emilsjolander** [comment]
Ok fixed: https://github.com/vislyhq/stretch/pull/15

**Comment by MichaelHoelzl** [comment]
> Ok fixed: #15

Wow, quick reaction and quick fix. 😃
